### ISSUE TITLE
1-step build process

### ISSUE THREAD

**Issue by jhsu802701** [OpenTwinCities/mnpay#20]
Create a custom Docker image and a build script, much like the opentwincities/rbenv-adoptatree Docker image and build_fast.sh script in the Adopt-a-Tree project.

**Comment by kjschiroo** [comment]
Could you explain more about what you would be aiming for here? I'm aspiring to keep the development and production environments as similar as possible.

It seems like you are recommending a single docker image where we are using multiple. Currently we use one docker image to serve our static files and another to serve dynamic pages. We will probably be adding another for database once we have enough data to stop using sqlite.

I think a single "docker-compose" build command would be good though. The only thing we are missing right now from this is spinning up an image to build `client.min.js`. It would be nice to have something to do that, because figuring out dependencies for that is a slight pain otherwise. 

Is that what you were thinking?

**Comment by jhsu802701** [comment]
I use Docker differently from most people.  I use Docker in a manner similar to that which I used for Vagrant.  I create a custom Docker image for this project.  In the command line, I enter the Docker container, download the source code, and then run a build_fast.sh script to execute the setup procedure.  I use a setup.sh script that starts up the server and outputs it to the IP address 0.0.0.0.  Then from my host machine, I can view the server output from the browser.

I can also use host system tools to view and edit the source code, which is kept in the shared directory.  (I have access to this shared directory through Docker and through my host system.)

**Comment by kjschiroo** [comment]
Take a look over at #23. This is the direction I'm thinking of going. We have separate containers for each service as well as a container for building the client. This makes the only dependencies that developers need to worry about docker and docker-compose. Does this meet what you are trying to accomplish? If not, how would you change it?

**Comment by jhsu802701** [comment]
The plan sounds good.  Django seems to be more popular than Flask.

I was able to get the app to work on my Mac with Docker Compose.  Now I have a better idea of how the more typical Docker solutions work.  My approach with Docker has been to use it as a replacement for Vagrant.  I've done some experimenting with a custom 64-bit Docker image and a custom 32-bit Docker image for MN Pay.  (I have both 64-bit and 32-bit versions for Adopt-a-Tree.  My desktop computer at home is 32 bits.)  I was able to get the "pip install -r requirements.txt" command to work with the 64-bit version but not in the 32-bit version.  Thus, it appears that at least one of the Python packages used requires a 64-bit system and cannot handle 32 bits.

For now, I'm putting this project on the back burner.  Getting the project to work under my Docker solution will be more work than I originally anticipated.  I'll focus on Adopt-a-Tree and other projects.
